### ISSUE TITLE
Timepicker events don't emit in real time

### ISSUE THREAD

**Issue by robertpataki** [ng-bootstrap/ng-bootstrap#3770]
### Bug description:
**The Timepicker component doesn't provide a way for the application to keep up with the time changes in real time:**

- The component's events aren't documented on its [API page](https://ng-bootstrap.github.io/#/components/timepicker/api#NgbTimepicker), although I've found that the component triggers both **(input)** and **(change)** events.
- When the user uses the ARROW UP and DOWN keys to change the input values, the **time** model changes internally, but nor the **input** nor the **change** events emit.
- The **input** event is triggered each time a keyboard key is pressed, but the time model doesn't get updated until the **change** event is triggered - which happens when the keyboard/mouse focus leaves the active input field
- Using the spinner buttons do change the **time** model, but no event emits - even when the keyboard focus leaves the active input field

### Link to minimally-working StackBlitz that reproduces the issue:

[https://stackblitz.com/edit/angular-ngb-timepicker-events](https://stackblitz.com/edit/angular-ngb-timepicker-events)

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular: 9.1.11

ng-bootstrap: 6.1.0

Bootstrap: 4.5.0

**Comment by benouat** [comment]
Hey @robertpataki 

I am not sure to understand exactly what you want to achieve. Even looking at your StakBlitz, it's not clear.
**TLDR**:  I would say that if you want to benefit from reactivity on this widget, your only option is to use `ngModel`. model will always be updated _live_ (both using inputs values, or the spinner icons).
Then if you want also to be notified, feel free to actually duplicate the `ngModel` usage

```html
<ngb-timepicker [(ngModel)]="time" (ngModelChange)="onLiveChange()" >
<!--
                    ↑                     ↑
            used for 2-way         be notified for each update
                bindings
--> 
```
--- 

Let me try to answer your statements

> The component's events aren't documented on its API page, although I've found that the component triggers both (input) and (change) events.

We don't do anything special internally regarding `input` and `change` events on all the timepicker input elements. So this is why you don't see anything in the doc, there is nothing to document!
Obviously, you can use these events, as they are standard events, and de-facto they will be fired, and will bubble.

> When the user uses the ARROW UP and DOWN keys to change the input values, the time model changes internally, but nor the input nor the change events emit.

 Again here, we don't do anything public regarding these events... so obviously there is nothing.

> The input event is triggered each time a keyboard key is pressed, but the time model doesn't get updated until the change event is triggered - which happens when the keyboard/mouse focus leaves the active input field

This has been implemented on purpose, this is a design choice

> Using the spinner buttons do change the time model, but no event emits - even when the keyboard focus leaves the active input field

Yes, you're right here. Keyboard management could be greatly improved on this widget.

**Comment by benouat** [comment]
@robertpataki were my explanations clear enough ? Can we close it if that's ok with you ?

**Comment by robertpataki** [comment]
Hi @benouat,

I apologise for being so late with my reply, I'm in the COVID-19 time vortex :/

Thanks so much for your detailed explanation. Looking at the ```(ngModelChange)``` event seem to solve every single problem I was having - and it's fair to say that all those problems were caused by the fact that I wrongly was trying to use native HTML element events instead of just listening to the model changes.

Thanks so much for the awesome components and the support.
